graph/kv/indexing.go

```diff
@@ -675,9 +675,10 @@ func (qs *QuadStore) hasPrimitive(ctx context.Context, tx BucketTx, p *proto.Pri
 	if !get && unique {
 		return p, nil
 	}
-	for _, x := range options {
+	for ix := len(options) - 1; ix >= 0; ix-- {
```

Comment:
Using `i` as a name is fine
```suggestion
	for i := len(options) - 1; i >= 0; i-- {
```